### PR TITLE
Mockito 5

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Use Mockito 5.

* Update Mockito to 5.12.0
* Removes use of the `mockito-inline` project as this is now the default
* Use the new package for the JUnit 4 runner
* Remove all `org.mockito.plugins.MockMaker` files since inline is now the default
* Removes PowerMock as this is not necessary anymore.
 
### Issues Resolved

N/A
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has a documentation issue. Please link to it in this PR.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by KarstenSchnitter — COMMENTED** [review]
Thanks for providing the change. To me, the main thing with that change is, that all builds and tests should be successful after merge. Unfortunately, the Gradle builds are failing as are some of the tests. Would be nice to see them in green to validate this change.

**Comment by dlvenable** [comment]
> Thanks for providing the change. To me, the main thing with that change is, that all builds and tests should be successful after merge. Unfortunately, the Gradle builds are failing as are some of the tests. Would be nice to see them in green to validate this change.

@KarstenSchnitter ,

Yes, I agree with getting the tests to green. I found some issues and resolved them. Now the tests are mostly to green. There are enough flaky tests to where I've not had all successful runs.

Also, the `Kafka plugin integration tests ` tests fail consistently.

File: data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/discovery/StaticPeerListProviderTest.java [hunk 1]
```diff
@@ -65,11 +67,12 @@ public void testListProviderWithNonEmptyList() {
...
-                new StringJoiner(MetricNames.DELIMITER).add(COMPONENT_SCOPE).add(COMPONENT_ID).add(PeerListProvider.PEER_ENDPOINTS).toString());
-        assertEquals(1, endpointsMeasures.size());
-        final Measurement endpointsMeasure = endpointsMeasures.get(0);
-        assertEquals(2.0, endpointsMeasure.getValue(), 0);
+        final ArgumentCaptor<ToDoubleFunction<List<String>>> gaugeFunctionCaptor = ArgumentCaptor.forClass(ToDoubleFunction.class);
+        verify(pluginMetrics).gauge(eq(PEER_ENDPOINTS), any(List.class), gaugeFunctionCaptor.capture());
+
+        final ToDoubleFunction<List<String>> gaugeFunction = gaugeFunctionCaptor.getValue();
+
+        assertThat(gaugeFunction.applyAsDouble(ENDPOINT_LIST), equalTo(2.0));
```
Comment (KarstenSchnitter) [inline data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/discovery/StaticPeerListProviderTest.java:75]: This looks somewhat strange. No call to `pluginMetrics.gauge(...)` is visible. For me, this makes the test a little harder to understand.

File: data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/discovery/DnsPeerListProviderTest.java [hunk 2]
```diff
@@ -90,17 +90,27 @@ public void testGetPeerList() {
...
+    public void testActivePeerCounter_with_single() {
         when(dnsAddressEndpointGroup.endpoints()).thenReturn(Collections.singletonList(Endpoint.of(ENDPOINT_1)));
-        assertEquals(1.0, endpointsMeasure.getValue(), 0);
+
+        final ArgumentCaptor<ToDoubleFunction<DnsAddressEndpointGroup>> gaugeFunctionCaptor = ArgumentCaptor.forClass(ToDoubleFunction.class);
+        verify(pluginMetrics).gauge(eq(PEER_ENDPOINTS), eq(dnsAddressEndpointGroup), gaugeFunctionCaptor.capture());
+
+        final ToDoubleFunction<DnsAddressEndpointGroup> gaugeFunction = gaugeFunctionCaptor.getValue();
+
+        assertThat(gaugeFunction.applyAsDouble(dnsAddressEndpointGroup), equalTo(1.0));
```
Comment (KarstenSchnitter) [inline data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/discovery/DnsPeerListProviderTest.java:113]: This looks somewhat strange. No call to `pluginMetrics.gauge(...)` is visible. For me, this makes the test a little harder to understand.

File: data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/discovery/DnsPeerListProviderTest.java [hunk 3]
```diff
@@ -90,17 +90,27 @@ public void testGetPeerList() {
...
-                .add(PeerListProvider.PEER_ENDPOINTS).toString());
-        assertEquals(1, endpointsMeasures.size());
-        final Measurement endpointsMeasure = endpointsMeasures.get(0);
-        assertEquals(2.0, endpointsMeasure.getValue(), 0);
+        final ArgumentCaptor<ToDoubleFunction<DnsAddressEndpointGroup>> gaugeFunctionCaptor = ArgumentCaptor.forClass(ToDoubleFunction.class);
+        verify(pluginMetrics).gauge(eq(PEER_ENDPOINTS), eq(dnsAddressEndpointGroup), gaugeFunctionCaptor.capture());
+
+        final ToDoubleFunction<DnsAddressEndpointGroup> gaugeFunction = gaugeFunctionCaptor.getValue();
 
+        assertThat(gaugeFunction.applyAsDouble(dnsAddressEndpointGroup), equalTo(2.0));
```
Comment (KarstenSchnitter) [inline data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/discovery/DnsPeerListProviderTest.java:101]: This looks somewhat strange. No call to `pluginMetrics.gauge(...)` is visible. For me, this makes the test a little harder to understand.

**Review by KarstenSchnitter — COMMENTED** [review]
Great to see the tests are green now. There are three places, where the tests now only consist of Mockito configuration without any directly apparent call to production code. Not sure, whether this can be improved.

**Comment by dlvenable** [comment]
@KarstenSchnitter , 

> Great to see the tests are green now.

Yes, me too! We still have a number of flaky tests.

I created #4723 to resolve a flaky test in the new AWS Lambda sink. This one was failing 18 out of 20 times when I locally set it to be a `@RepeatedTest`. Feel free to take a look at that one as well.

**Review by KarstenSchnitter — APPROVED** [review]
I think, this is fine to go.

**Pull request merged**